pysph/tools/binder.py

```diff
@@ -0,0 +1,208 @@
+    ]
+
+    nbf.write(nb, path+'/'+sim_name+'.ipynb')
+    return
+
+
+def find_sim_dirs(path, sim_paths_list=[]):
+    '''
+    Finds all the directories in a given directory that
+    contain pysph output files.
+    '''
+    path = os.path.abspath(path)
+    sim_files = get_files(path)
+    if len(sim_files) != 0:
+        sim_paths_list.append(path)
+    elif len(sim_files) == 0:
+        files = os.listdir(path)
+        files = [f for f in files if os.path.isdir(f)]
+        files = [os.path.abspath(f) for f in files if not f.startswith('.')]
+        for f in files:
+            sim_paths_list = find_sim_dirs(f, sim_paths_list)
+
+    return sim_paths_list
+
+
+def find_dir_size(path):
+    '''
+    Finds the size of a given directory.
+    '''
+    total_size = 0
+    for dir_path, dir_names, file_names in os.walk(path):
+        for f in file_names:
+            fp = os.path.join(dir_path, f)
+            if not os.path.islink(fp):
+                total_size += os.path.getsize(fp)
+
+    return total_size
+
+
+def make_binder(path):
+
+    src_path = os.path.abspath(path)
+
+    sim_paths_list = find_sim_dirs(src_path)
+
+    for path in sim_paths_list:
+        sim_name = os.path.split(path)[1]
+        make_notebook(path, sim_name)
+        if len(sim_paths_list) == 1 and sim_paths_list[0] == src_path:
+            files = os.listdir(src_path)
+            files = [src_path+'/'+f for f in files]
+            mkdir(src_path+'/'+sim_name)
+            could_not_copy = []
+            for f in files:
+                try:
+                    copy(f, src_path+'/'+sim_name)
+                except:
+                    could_not_copy.append(f)
+                    continue
+                os.remove(f)
+            if len(could_not_copy) != 0:
+                print("Could not copy the following files:\n")
+                for f in could_not_copy:
+                    print(f+'\n')
+
+    with open(src_path+'/requirements.txt', 'w') as file:
+        file.write(
+            "ipympl\n" +
+            "matplotlib\n" +
+            "ipyvolume\n" +
+            "numpy\n" +
+            "pytools\n" +
+            "-e git+https://github.com/pypr/compyle#egg=compyle\n" +
+            "-e git+https://github.com/pypr/pysph#egg=pysph"
+        )
+
+    with open(src_path+'/README.md', 'w') as file:
+        file.write(
+            "# Title\n" +
+            "[![Binder](https://mybinder.org/badge_logo.svg)]" +
+            "(https://mybinder.org/v2/[hosting_service]/[repo_specifics]" +
+            "/[branch_name])\n" +
+            "\n" +
+            "[comment]: # (GitLab users: hosting_service = gl, " +
+            "repo_specifics = [user_name]%2F[repo_name])\n" +
+            "\n" +
+            "[comment]: # (GitHub users: hosting_service = gh, " +
+            "repo_specifics = [user_name]/[repo_name])\n" +
+            "\n" +
+            "[comment]: # (Gist users: hosting_service = gist, " +
```

Comment:
Instead of this comment, can you just add all three examples and have users delete/modify those lines:
```
![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gl/user_name/repo_name/master)
```
Or whatever is needed. For more cases, users can always read some documentation -- perhaps a link to the binder docs may be useful for them to quickly refer instead of having to google it.

Reply:
I've changed dummy link to work for GitHub repos only, and added a statement saying that the user can go to mybinder.org for information on working with other hosting services.

Reply:
Can you use https://mybinder.readthedocs.io instead of just mybinder.org so it is easier to get to.

Reply:
Changed it in the latest commit.